src/gcp/cloudscheduler.ts

```diff
@@ -12,50 +13,109 @@ export interface Schedule {
     auth: true,
     origin: api.cloudschedulerOrigin,
   });
 }
 
+/*
+* getJob gets a cloudScheduler job with the given name
+* @returns a response from the cloud scheduler api
+* If no job with that name exists, this will return a 404
+*/
 export function getJob(name: string): Promise<any> {
   return api.request("GET", `/${VERSION}/${name}`, {
     auth: true,
     origin: api.cloudschedulerOrigin,
   });
 }
 
-export async function createOrReplaceJob(schedule: Schedule): Promise<void> {
-  const jobName = `${schedule.name.split("/")[5]}`;
+/*
+* updateJob updates a cloudScheduler job
+* @returns a response from the cloud scheduler api
+* If no job with that name exists, this will return a 404
+*/
+export function updateJob(job: Job): Promise<any> {
+  return api.request("PATCH", `/${VERSION}/${job.name}`, {
+    auth: true,
+    origin: api.cloudschedulerOrigin,
+    data: Object.assign({ timeZone: "America/Los_Angeles" }, job),
+  });
+}
+
+/*
+* createOrReplaceJob checks for a existing job with the given name
+* if none is found, it creates a new job
+* if one is found, and it is identical to the job parameter, it does nothing
+* otherwise, if one is found and it is different from the job param, it updates the job
+*/
+export async function createOrReplaceJob(job: Job): Promise<any> {
+  const jobName = `${job.name.split("/")[5]}`;
   try {
-    await createJob(schedule);
-    logLabeledSuccess("functions", `created scheduler job ${jobName}`);
-  } catch (e) {
-    if (e.context.response.statusCode !== 409) {
-      throw e;
+    const existingJob = await getJob(job.name);
+    if (!job.timeZone) {
+      // We set this here to avoid recreating schedules that use the default timeZone
+      job.timeZone = "America/Los_Angeles";
     }
-    const existingJob = await getJob(schedule.name);
-    if (existingJob.body.schedule !== schedule.schedule) {
-      logLabeledBullet("functions", `re-creating scheduler job ${jobName}`);
-      await deleteJob(schedule.name);
-      return createJob(schedule);
+
+    if (_isIdentical(existingJob.body, job)) {
+      logLabeledBullet("functions", `scheduler job ${jobName} is up to date, no changes required`);
+      return;
     } else {
-      logLabeledBullet("functions", `keeping scheduler job ${jobName} since it has not changed`);
+      logLabeledBullet("functions", `updating scheduler job ${jobName}`);
+      return updateJob(job);
+    }
+  } catch (e) {
+    // if the error status is 404, no job exists, so we can create one
+    // if it is anything else, we should error out
+    if (e && e.context && e.context.response && e.context.response.statusCode !== 404) {
+      throw e;
     }
+    await createJob(job);
+    logLabeledSuccess("functions", `created scheduler job ${jobName}`);
   }
 }
+
+/*
+* _isIdentical is a helper function to check if 2 jobs are funcitonally equivalent
+*/
+function _isIdentical(existingJob: Job, newJob: Job): boolean {
```

Comment:
Don't need `_` in ts to signal private function. Jsdoc also needs to be consistent with the rest.